sdks/python/apache_beam/io/range_trackers.py

```diff
@@ -324,3 +324,123 @@ def set_current_position(self, position):
 
   def fraction_consumed(self):
     return self._range_tracker.fraction_consumed()
+
+
+class LexicographicKeyRangeTracker(iobase.RangeTracker):
+  """
+  A range tracker that tracks progress through a lexicographically
+  ordered keyspace of strings.
+  """
+
+  def __init__(self, start_key=None, stop_key=None):
+    self._start_key = start_key
+    self._stop_key = stop_key
+    self._lock = threading.Lock()
+    self._last_claim = None
+
+  def start_position(self):
+    return self._start_key
+
+  def stop_position(self):
```

Comment:
Perhaps I'm misunderstanding something (is Python's memory model very different from C++/Java?), but stop_key is being modified from a different thread, it should be written under a lock (it is) and read under a lock (it isn't). What am I missing?